src/coreclr/src/jit/lsrabuild.cpp

```diff
@@ -2925,14 +2925,23 @@ int LinearScan::BuildStoreLoc(GenTreeLclVarCommon* storeLoc)
         // srcCount = number of registers in which the value is returned by call
         GenTreeCall*    call        = op1->AsCall();
         ReturnTypeDesc* retTypeDesc = call->GetReturnTypeDesc();
-        unsigned        regCount    = retTypeDesc->GetReturnRegCount();
         srcCount                    = retTypeDesc->GetReturnRegCount();
 
         for (int i = 0; i < srcCount; ++i)
         {
             BuildUse(op1, RBM_NONE, i);
         }
     }
+    else if (op1->isContained() && op1->OperIs(GT_BITCAST))
+    {
+        GenTree*     bitCastSrc   = op1->gtGetOp1();
+        RegisterType registerType = bitCastSrc->TypeGet();
+        singleUseRef              = BuildUse(bitCastSrc, allRegs(registerType));
+
+        Interval* srcInterval = singleUseRef->getInterval();
+        assert(srcInterval->registerType == registerType);
+        srcCount = 1;
```

Comment:
> Is it the only case where we increment srcCount for a contained operand?

No, this is not the only case where we have a non-zero srcCount for a contained operand. A load with a base and no index would also have a srcCount of 1, but addressing modes are handled specially so it looks a little different in that case.

The only place we have a contained `BITCAST` is when its source is neither contained nor reg-optional. In that case it must produce a result, and therefore we need `srcCount` to be 1. If `bitCastSrc` is (or may be) on the stack it will either be contained or reg-optional.